### ISSUE TITLE
OACMPDM1 glitching when standing near to the teleporter.

### ISSUE THREAD

**Issue by The-Gig** [NeonKnightOA/oacmpvol1#24]
Testing with z_oacmp1v4_beta-220411.pk3.
This version introduced weird glitching when standing near to the central teleporter. I don't know why. It looks like it wasn't there in the previous version.
![shot0175](https://user-images.githubusercontent.com/23258514/162693457-43e2b1cb-3a45-4022-80d7-5df101c0fd6e.jpg)

**Comment by EddieBrrrock** [comment]
Look at my others issues :
https://github.com/NeonKnightOA/oacmpvol1/issues/28
https://github.com/NeonKnightOA/oacmpvol1/issues/29
I just noticed they are all related to the left side of the teleporter (or the right side for the liz statue)

**Comment by The-Gig** [comment]
I guess all of them might come from a single small thing, like a caulk brush visible to the user or something like that...

**Comment by EddieBrrrock** [comment]
maybe that ? :
![Capture d’écran 2022-04-12 à 11 24 38](https://user-images.githubusercontent.com/99912467/162927792-5664e1b7-e462-433a-acaf-ac266c1aca60.png)
![Capture d’écran 2022-04-12 à 11 24 35](https://user-images.githubusercontent.com/99912467/162927851-5c41d93b-577e-4dc8-b9cf-6eba7b73fbf4.png)

**Comment by EddieBrrrock** [comment]
because it cause problems even when we are far from the tp
![Capture d’écran 2022-04-12 à 11 24 10](https://user-images.githubusercontent.com/99912467/162928241-64b0252f-5eec-4f91-a8c2-440e0ae6fcb5.png)

**Comment by The-Gig** [comment]
I opened the current version of the map in Radiant (let's say corresponding to 220411 pk3) and so far the only odd thing I noticed is what looks like missing "clusterportal" brushes at the sides of the teleporter (the should extend from the yellow botclip to the wall).
![immagine](https://user-images.githubusercontent.com/23258514/162940168-ae06ba11-9b7d-4d3d-974e-e0e7a163d15c.png)
But I thought in theory clusterportals should only affect bots behavior, IDK whether they can also be connected to visual glitches for the player...

**Comment by NeonKnightOA** [comment]
Yeah, can't remember where I left the building script I used with this map so I've compiled it with the default script that I use. I need to keep checking.

**Comment by The-Gig** [comment]
I don't think it's the building script. I have a feeling that when doing the small edits here <https://github.com/NeonKnightOA/oacmpvol1/commit/c51ca2a5cb3376798a74b8ed033c5f4f00875e82?diff=unified> you "selected" a few hint brushes for some reason and the editor "snapped them to grid", as you can see in the diff it looks like some brush coordinates got their fractional part removed.
After disabling "View->Filter->Translucent", I see the map has got a lot of (at least apparently) oddly angles hint brushes. And some of them look huge but are actually split into smaller brushes.
To provide some context:
Here it is version e1f552b (from 2017) with Translucent filtered out:
![immagine](https://user-images.githubusercontent.com/23258514/163990078-5e33de4d-9d17-4017-bd3e-6307b9fe086f.png)
And it is the same version with Translucent shown:
![immagine](https://user-images.githubusercontent.com/23258514/163990251-dfd09818-a37f-408c-a39b-fc5f1015ed1f.png)
In the middle, you can see highlighted in red _brush 1183_, one of those modified in the commit c51ca2a (from 2022-04-08, the one which introduced the glitch). Let's look at it closely.
This is the e1f552b (2017) version
![immagine](https://user-images.githubusercontent.com/23258514/163990780-3ce5ee47-9484-4428-ab0c-8e8d4a00b451.png)
And this is an even closer look. Shown grid size is 1.
![immagine](https://user-images.githubusercontent.com/23258514/163991141-3a0404e0-42a2-472d-a002-a2faf36059e0.png)
Let's do the same with c51ca2a (2022-04-08):
![immagine](https://user-images.githubusercontent.com/23258514/163991313-80cbcb99-a3fb-4752-9dba-0e84dddb7a23.png)
and looking closer:
![immagine](https://user-images.githubusercontent.com/23258514/163991473-103c62ea-8a16-4c25-aabd-40652aca376d.png)
See? Now that specific brush is snapped to grid, it doesn't perfecly align anymore to the other hint brushes next to it (which aren't snapped), despite looking like it is at first glance.
![immagine](https://user-images.githubusercontent.com/23258514/163991856-97b5c4e8-d301-4e34-a401-f55acdeab8f7.png)
So maybe it's this misalignment between adjacent hint brushes that makes VIS optimization go crazy?

I don't know the reason why Akom did those hint brushes that way, I guess optimizing a map like this may require some trickery.

I wonder whether it may be easier to manually fix those hint brushes or to restart from 2017 version e1f552b (https://github.com/NeonKnightOA/oacmpvol1/blob/e1f552be3f4c277b93c7175520c8c978906d5c5e/sources/maps/oacmpdm1.map) and redo the changes you did in the two commits from past weeks (https://github.com/NeonKnightOA/oacmpvol1/commit/c51ca2a5cb3376798a74b8ed033c5f4f00875e82 and https://github.com/NeonKnightOA/oacmpvol1/commit/8b911473490e2313131ce5865ddb6f3b13acb8e7) without touching the hint brushes?

**Comment by NeonKnightOA** [comment]
May be related with the absence of common/skip. I'm recompiling all the maps now that it's included and see how many problems that addition solve.

**Comment by NeonKnightOA** [comment]
I've gotten rid of the hint/skip brushes and the problem got solved. Still needs testing.

**Comment by The-Gig** [comment]
I think I managed to restore working hint brushes: see https://github.com/NeonKnightOA/oacmpvol1/issues/29#issuecomment-1469972860

**Comment by NeonKnightOA** [comment]
https://mega.nz/file/bMZkRJBa#J_rUm2nz2SpiMFl6kBY0eMceeCLOJ1PLfnz_-CQRzCw
New test version.
